python/taichi/lang/util.py

```diff
@@ -191,6 +191,8 @@ def cook_dtype(dtype):
         return dtype
     if isinstance(dtype, _ti_core.Type):
         return _ti_core.DataType(dtype)
+    if isinstance(dtype, impl.MatrixType):
```

Comment:
`cook_dtype` is used for transformations like `float` -> `default_fp`. Why declare a tensor type in C++ here?

Reply:
This function also transforms python_type to c++_type here. I think this is needed.